app/bundles/LeadBundle/Controller/Api/LeadApiController.php

```diff
@@ -205,6 +205,106 @@ public function getNotesAction($id)
     }
 
     /**
+     * Obtains DonotContact status on a specific Contact
+     *
+     * @param $id
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function getDoNotEmailAction($id)
+    {
+        $entity = $this->model->getEntity($id);
+        if ($entity !== null) {
+            if (!$this->factory->getSecurity()->hasEntityAccess('lead:leads:viewown', 'lead:leads:viewother', $entity->getOwner())) {
```

Comment:
`$this->factory` is now deprecated and shouldn't be used. Get the service from the container instead.

For example, `$this->factory->getSecurity()` would be `$this->get('mautic.security')`; `$this->factory->getEntityManager()` would be `$this->get('doctrine')->getManager()->getRepository('MauticLeadBundle:Lead')` or `$this->getModel('lead')->getRepository()`.

Same for the other uses of $this->factory later in this PR